pkg/csi/manager.go

```diff
@@ -70,8 +76,39 @@ func (m *Manager) Run(cfg *config.Config) error {
 		return err
 	}
 
+	nodeID := cfg.NodeID
+
+	ifaces, err := sysfsnet.Interfaces()
+	if err != nil {
+		logrus.WithError(err).Warn("Failed to enumerate MAC addresses for VMI discovery")
+	}
+
+	name, err := discoverVMIName(virtClient.VirtualMachineInstance(namespace), ifaces)
+	if err == nil {
+		nodeID = name
+		logrus.WithFields(logrus.Fields{
+			"node_id_original":   cfg.NodeID,
+			"node_id_discovered": name,
+		}).Info("Discovered Harvester VM node ID")
+	}
+	if errors.Is(err, errVMINotFound) {
+		var readableIfaces []string
+		for _, i := range ifaces {
+			readableIfaces = append(readableIfaces, i.HardwareAddr.String())
+		}
+		logrus.WithFields(logrus.Fields{
+			"namespace":     namespace,
+			"mac_addresses": readableIfaces,
+			"hostname":      cfg.NodeID,
+		}).Warn("Did not find any VMIs that match this node's MAC addresses; falling back to hostname as VMI name")
+		err = nil
```

Comment:
Don't recommend setting err as nil here. It's not clear and readable. Prefer to use `if...else if ...else` or `switch...case` to distinguish the error.

Reply:
Went with if...else if..else for the next revision 🙂